sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetColumnVector.java

```diff
@@ -84,6 +84,8 @@ final class ParquetColumnVector {
       if (defaultValue == null) {
         vector.setAllNull();
         return;
+      } else {
+        vector.setHasDefaultValue();
```

Comment:
is it an existing bug?

Reply:
Before this PR, there is no existing bug, after this PR, for Parquet tables whose columns have associated DEFAULT values, the result may be incorrect. 
https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetColumnVector.java#L84C15-L100

So I add a flag `protected boolean hasDefaultValue = false;` to indicate if the current column vector has default value and not to clean the data if it's true.